### PR TITLE
Adding security context to initContainer

### PR DESCRIPTION
**Pull request by kndoni**
**Changes** 

1. Adding security context to init container as keyverno policies implies that initContainer should have securityContext defined as well
2. Adding resources for every container and initContainer on each daemonset.
3. Adding property devicePluginMps in values that controlles enabling or disabling mps daemonset.

### CODE REVIEW

File: deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml [hunk 1]
```diff
@@ -92,6 +92,8 @@ spec:
             mountPath: /available-configs
           - name: config
             mountPath: /config
+        securityContext:
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml:95]: @klueska @cdesiniotis do you have any specifics on why this was omitted?

**Review by elezar — CHANGES_REQUESTED** [review]
@kndoni please also add this to the other daemonsest in the helm package.

**Comment by kndoni** [comment]
> @kndoni please also add this to the other daemonsest in the helm package.

I have added security context for other containers in daemonsets as well

**Review by elezar — APPROVED** [review]

File: deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml [hunk 2]
```diff
@@ -61,6 +61,7 @@ spec:
         command: [mps-control-daemon, mount-shm]
         securityContext:
           privileged: true
+          {{- include "mps-control-daemon.securityContext" . | nindent 10 }}
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml:64]: Note that this helper template is not defined.

File: deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml [hunk 3]
```diff
@@ -169,6 +172,7 @@ spec:
             value: compute,utility
           securityContext:
             privileged: true
+            {{- include "mps-control-daemon.securityContext" . | nindent 10 }}
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml:175]: Since we're already running the container as `privileged`, I don't think this is relevant here.

**Review by elezar — CHANGES_REQUESTED** [review]
I don't think that adding a security context if we're already running a container as priveleved makes sense. We also do not have the template used for the mps-control-daemon defined.

File: deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml (hunk 2 above)
Comment (kndoni) [reply deployments/helm/nvidia-device-plugin/templates/daemonset-mps-control-daemon.yml]: Removed this as well

**Comment by kndoni** [comment]
@elezar Let me know if something else should be changed as well.
Thanks in advance!

**Review by elezar — CHANGES_REQUESTED** [review]
@kndoni please remove the changes to the MPS API from this PR.

I'm not opposed to making the change, but would like to separate the security-context specific changes from the API changes.

File: deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml [hunk 4]
```diff
@@ -92,6 +92,12 @@ spec:
             mountPath: /available-configs
           - name: config
             mountPath: /config
+        {{- with .Values.resources }}
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml:95]: A general question: Does it make sense to apply the same resources to each of the containers?

File: deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml [hunk 5]
```diff
@@ -184,6 +194,9 @@ spec:
             value: compute,utility
         securityContext:
           {{- include "nvidia-device-plugin.securityContext" . | nindent 10 }}
+          capabilities:
+            add:
+              - SYS_ADMIN
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-device-plugin.yml:199]: Why was this added? There was an issue #856 which should be resolved in the latest release.

File: deployments/helm/nvidia-device-plugin/templates/daemonset-gfd.yml [hunk 6]
```diff
@@ -175,7 +185,10 @@ spec:
           - name: GFD_MODE
             value: "{{ .Values.gfdMode }}"
         securityContext:
-          {{- include "gpu-feature-discovery.securityContext" . | nindent 10 }}
+          {{- include "nvidia-device-plugin.securityContext" . | nindent 10 }}
+          capabilities:
+            add:
+              - SYS_ADMIN
```
Comment (elezar) [inline deployments/helm/nvidia-device-plugin/templates/daemonset-gfd.yml:191]: Similar question here. See #856
